x-pack/legacy/plugins/index_management/public/app/services/documentation.ts

```diff
@@ -33,6 +35,18 @@ class DocumentationService {
   public getIdxMgmtDocumentationLink() {
     return `${this.kibanaDocsBase}/managing-indices.html`;
   }
+
+  public getTypeDocLink = (type: DataType): string | undefined => {
+    const typeDefinition = TYPE_DEFINITION[type];
+    if (!typeDefinition || !typeDefinition.docUri) {
+      return undefined;
+    }
+    return `${this.esDocsBase}${typeDefinition.docUri}`;
+  };
+
+  public getGeoShapeDocumentationLink() {
```

Comment:
I would prefer if possible to keep a single source of truth (a single place) for the type definition and documentation URLs.
To support multiple URIs, what do you think about changing the Type to

```js
export interface DataTypeDefinition {
  label: string;
  value: DataType;
  documentation?: {
    main: string;
    [key: string]: string;
  };
  subTypes?: { label: string; types: SubType[] };
}
```

And then declare `geo_shape` this way

```js
geo_shape: {
  label: 'Geo shape',
  value: 'geo_shape',
  documentation: {
    main: '/geo-shape.html',
    learMore: '/geo-shape.html#geoshape-indexing-approach',
  },
},

// and getTypeDocLink would be

public getTypeDocLink = (type: DataType, uri = 'main'): string | undefined => {
  const typeDefinition = TYPE_DEFINITION[type];
  if (!typeDefinition || !typeDefinition.documentation || !typeDefinition.documentation[uri]) {
    return undefined;
  }
  return `${this.esDocsBase}${typeDefinition.documentation[uri]}`;
};
```